src/test/scala/is/hail/asm4s/ASM4SSuite.scala

```diff
@@ -337,4 +337,32 @@ class ASM4SSuite extends TestNGSuite {
     assert(f(2,2,8) == 16)
   }
 
+
+  @Test def checkLocalVarsOnMethods(): Unit = {
+    val fb = FunctionBuilder.functionBuilder[Int, Int, Int, Int]
+    val add = fb.newMethod[Int, Int, Int]
+    val sub = fb.newMethod[Int, Int, Int]
+    val mult = fb.newMethod[Int, Int, Int]
+
+    val v1 = add.newLocal[Int]
+    val v2 = add.newLocal[Int]
+
+    add.emit(Code(v1 := add.getArg[Int](1),
+      v2 := add.getArg[Int](2),
+      v1 + v2))
+    sub.emit(sub.getArg[Int](1) - sub.getArg[Int](2))
```

Comment:
I'm going to get rid of `sub` + `mul` since they don't actually add anything to the test. This is just to check that assigning localrefs on a method isn't messed up.